exporter/awsemfexporter/metric_translator.go

```diff
@@ -404,12 +414,15 @@ func calculateRate(fields map[string]interface{}, val interface{}, timestamp int
 	return metricRate
 }
 
-func dimensionRollup(dimensionRollupOption string, originalDimensionSlice []string) [][]string {
+func dimensionRollup(dimensionRollupOption string, originalDimensionSlice []string, instrumentationLibName string) [][]string {
 	var rollupDimensionArray [][]string
-	dimensionZero := []string{OtlibDimensionKey}
+	var dimensionZero []string
+	if instrumentationLibName != noInstrumentationLibraryName {
+		dimensionZero = append(dimensionZero, OTellibDimensionKey)
+	}
 	if dimensionRollupOption == ZeroAndSingleDimensionRollup {
 		//"Zero" dimension rollup
-		if len(originalDimensionSlice) > 0 {
+		if len(originalDimensionSlice) > 0 && len(dimensionZero) > 0 {
 			rollupDimensionArray = append(rollupDimensionArray, dimensionZero)
 		}
```

Comment:
What will happen if one metric has several dimensions but do not have instrumentationLibName defined? Can it have Zero dimension rollup?
For example, `Metric1` has **Dimension** `A` and `B`, but do not have `instrumentationLibName`. If customer enables `ZeroDimensionRollup`, will he still get the `Metric1` without any dimension? It looks like you didn't append `[]` into `rollupDimensionArray` if `instrumentationLibName == noInstrumentationLibraryName`. Or is this something we expect?

Reply:
Yes, we should generate zero dimension metric for this case. Made the update and added test cases to cover these combinations. Thanks.